database/dtb.ksy

```diff
@@ -0,0 +1,127 @@
+meta:
+  id: dtb
+  title: Flattened Devicetree Format
+  file-extension: dtb
+  application:
+    - Linux
+    - Das U-Boot
+  xref:
+    wikidata: Q16960371
+  license: CC0-1.0
+  encoding: ASCII
+  endian: be
+doc: |
+  Also referred to as Devicetree Blob (DTB). It is a flat
+  binary encoding of data (primarily devicetree data, although
+  other data is possible as well).
+
+  On Linux systems that support this the blobs can be accessed in
+  /sys/firmware/fdt :
+
+  - https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-firmware-ofw
+
+  The encoding of strings used in the strings block and struct block is
+  actually a subset of ASCII:
+
+  https://github.com/devicetree-org/devicetree-specification/blob/master/source/chapter2-devicetree-basics.rst
+
+  Example files:
+
+  - https://github.com/qemu/qemu/tree/master/pc-bios
+doc-ref:
+  - https://github.com/devicetree-org/devicetree-specification/releases/tag/v0.3
+  - https://github.com/devicetree-org/devicetree-specification/blob/ba2aa67/source/flattened-format.rst
+  - https://elinux.org/images/f/f4/Elc2013_Fernandes.pdf
+seq:
+  - id: magic
+    -orig-id: magic
+    contents: [0xd0, 0x0d, 0xfe, 0xed]
+  - id: total_size
+    -orig-id: totalsize
+    type: u4
+  - id: structure_block_offset
+    -orig-id: off_dt_struct
+    type: u4
+  - id: strings_block_offset
+    -orig-id: off_dt_strings
+    type: u4
+  - id: memory_reservation_block_offset
+    -orig-id: off_mem_rsvmap
+    type: u4
```

Comment:
According to the KSY Style Guide (https://doc.kaitai.io/ksy_style_guide.html#attr-id), these three fields and also this one:

https://github.com/kaitai-io/kaitai_struct_formats/blob/7fc9effd35ff7106f3e52ed1746e7a075068e200/database/dtb.ksy#L114-L116

should use the prefix `ofs_`:

> For fields that designate **offset** to some particular data structure, use `ofs_` prefix and name of that data structure (as it would appear in the file) — i.e. `ofs_block`, `ofs_queries`, `ofs_path`

so:

```shell
sed -i 's/structure_block_offset/ofs_structure_block/g' dtb.ksy
sed -i 's/strings_block_offset/ofs_strings_block/g' dtb.ksy
sed -i 's/memory_reservation_block_offset/ofs_memory_reservation_block/g' dtb.ksy
sed -i 's/name_offset/ofs_name/g' dtb.ksy
```